### DISCUSSION TITLE
[나래] assignment 기능구현

### DISCUSSION THREAD

**Discussion by NR0617** [Wanted-07-team-9/pre-onboarding-7th-2-1-9#20] [기능구현]
배포 링크 : https://imaginative-donut-17d06f.netlify.app/
<img src="https://img.shields.io/badge/javascript-F7DF1E?style=for-the-badge&logo=javascript&logoColor=black"> <img src="https://img.shields.io/badge/react-61DAFB?style=for-the-badge&logo=react&logoColor=black"> <img src="https://img.shields.io/badge/styledcomponents-DB7093?style=for-the-badge&logo=styled-components&logoColor=white">

# Assignment 1
### 디테일 페이지
1. 디테일 페이지 컴포넌트
https://github.com/Wanted-07-team-9/pre-onboarding-7th-2-1-9/blob/dfd7a0f6ad08f8522a74ddf4e079ebc8e92c4952/src/pages/DetailPage/index.js#L10-L28

2. 데이터 요청 함수
https://github.com/Wanted-07-team-9/pre-onboarding-7th-2-1-9/blob/dfd7a0f6ad08f8522a74ddf4e079ebc8e92c4952/src/api/fetchData.js#L3-L11

3. 데이터 필터링 버튼(헤더)
https://github.com/Wanted-07-team-9/pre-onboarding-7th-2-1-9/blob/dfd7a0f6ad08f8522a74ddf4e079ebc8e92c4952/src/components/Headers/AttributeHeader.js#L1-L23

# Assignment 2
### 필수 요구 사항
 1. Figma 상의 디자인 및 기능 구현
 
![showApp](https://user-images.githubusercontent.com/94212747/200127967-984ea8c8-b886-4b4b-994a-d958a2aad12b.gif)

**Comment by songseul** [comment]
- api가 하나여서 상세페이지를 필터로 구현 하신 점이 좋은것 같아용!

**Comment by jun-05** [comment]
- 컴포넌트에서 3항 연산자를 너무 많이 사용하시는것 같다는 생각이 들었어요.
- if(loading) return <loading/> 형식으로 해보는게 좋지 않을까 싶었습니다!
- 배포는 ci/cd 안해도 aws s3 만들고 그냥 빌드 안에 있는 파일 붙여넣으면 되더라구요. 다음엔 배포까지 파이팅!!

**Comment by crystal993** [comment]
- Detail 페이지에 useEffect 안에서 getDatas() 로직을 추상화 시키면 좋을 것 같다는 생각이 들었습니다. 
- 태그 버튼들을 Object.keys로 가져온 점이 인상깊습니다. 저는 참조해서 가져왔는데 다음에는 저도 적용해보고 싶네요!!

**Comment by tirhande** [comment]
- getCarsData에서 payload 를 filter로 구분하신게 좋아요!
- axios 요청 보낼때 param을 config 형태로 붙이시면 더 좋을것 같아요!
